### ISSUE TITLE
Cannot work with BrainpoolP256r1 Key and Certificate

### ISSUE THREAD

**Issue by vbrandl** [rustls/rustls#1332]
I just tried executing the `tlsserver-mio` example using a BrainpoolP256r1 private key and certificate. The curve is listed in the [NamedCurve](https://docs.rs/rustls/latest/rustls/internal/msgs/enums/enum.NamedCurve.html#variant.brainpoolp256r1) enum.

When starting, the server panics:

```
$ cargo run -- --certs cert.pem --key key.pem http
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/tlsproxy --certs cert.pem --key key.pem http`
thread 'main' panicked at 'bad certificates/private key: General("invalid private key")', src/main.rs:598:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Is this a bug or are there no plans to support brainpool keys? I could observe this in multiple languages: Java dropped support for brainpool and also golang cannot use the same key.

Attached you can find the test key.

[key_and_cert.zip](https://github.com/rustls/rustls/files/11906457/key_and_cert.zip)

**Comment by ctz** [comment]
Yes, we don't support any brainpool curves, or explicitly-defined curves. Though I'm aware they are mandated by a bunch of German national security standards, they are -- as I understand it -- not cryptographically better than the NIST curves.

**Comment by djc** [comment]
Might be useful to note that #1319 should eventually make it possible to support Brainpool curves with rustls (although someone would have to write an implementation for it).

**Comment by cpu** [comment]
> The curve is listed in the [NamedCurve](https://docs.rs/rustls/latest/rustls/internal/msgs/enums/enum.NamedCurve.html#variant.brainpoolp256r1) enum.

We've updated the Rustdoc's to help make it clearer that `NamedCurve` isn't the list of supported curves (https://github.com/rustls/rustls/pull/1333). Thanks for filing this issue!

**Comment by djc** [comment]
With that, I think we can close this?

**Comment by vbrandl** [comment]
Yeah, thanks for the pointer to the PR, I'll watch the progress on this one.
Also you are right. We must use brainpool due to German regulations...
